diff --git a/floating-ui-react/middleware/arrow.mjs b/floating-ui-react/middleware/arrow.mjs
--- a/floating-ui-react/middleware/arrow.mjs
+++ b/floating-ui-react/middleware/arrow.mjs
@@ -88,7 +88,8 @@ export const baseArrow = options => ({
  * This wraps the core `arrow` middleware to allow React refs as the element.
  * @see https://floating-ui.com/docs/arrow
  */
-export const arrow = (options, deps) => ({
-  ...baseArrow(options),
-  options: [options, deps]
-});
\ No newline at end of file
+export const arrow = (options, deps) => {
+  const middleware = baseArrow(options);
+  middleware.options = [options, deps];
+  return middleware;
+};
