diff --git a/dist/cli/build/utils/workerd.js b/dist/cli/build/utils/workerd.js
--- a/dist/cli/build/utils/workerd.js
+++ b/dist/cli/build/utils/workerd.js
@@ -43,12 +43,12 @@ export function transformBuildCondition(conditionMap, condition) {
 export function transformPackageJson(json) {
     const transformed = structuredClone(json);
     let hasBuildCondition = false;
-    if (json.exports) {
+    if (json.exports && typeof json.exports === "object" && !Array.isArray(json.exports)) {
         const exp = transformBuildCondition(json.exports, "workerd");
         transformed.exports = exp.transformedExports;
         hasBuildCondition ||= exp.hasBuildCondition;
     }
-    if (json.imports) {
+    if (json.imports && typeof json.imports === "object" && !Array.isArray(json.imports)) {
         const imp = transformBuildCondition(json.imports, "workerd");
         transformed.imports = imp.transformedExports;
         hasBuildCondition ||= imp.hasBuildCondition;
